File: SoloLearn Codes/Credit Card Validator/CreditCardValidator.java
import java.util.*;

public class CreditCardValidator
{
    public static void main(String[] args) {
        String candidate = new StringBuilder(new Scanner(System.in).nextLine()).reverse().toString();
        int[] digits = Arrays
                .stream(candidate.split(""))
                .mapToInt(Integer::parseInt)
                .toArray();
        for(int i = 1; i < digits.length; i += 2) {
            digits[i] *= 2;
            if(digits[i] > 9) digits[i] -= 9;
        }
        System.out.println(Arrays.stream(digits).sum() % 10 == 0 && digits.length == 16 ? "valid" : "not valid");
	}
}

File: SoloLearn Codes/Credit Card Validator/credit_card_validator.rb
s = gets.chomp.reverse
puts(s.chars.map.with_index{ |c, i| i % 2 == 1 ? c.to_i * 2 : c.to_i }.map{ |x| x > 9 ? x - 9 : x }.sum % 10 == 0 && s.length == 16 ? "valid" : "not valid")